sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_base_client.py

```diff
@@ -3,75 +3,38 @@
-            credential_policy = CognitiveServicesCredentialPolicy(credential)
-        elif credential is not None:
-            raise TypeError("Unsupported credential: {}. Use an instance of TextAnalyticsApiKeyCredential "
-                            "or a token credential from azure.identity".format(type(credential)))
+def _authentication_policy(credential):
+    credential_policy = None
+    if credential is None:
+        raise ValueError("Parameter 'credential' must not be None.")
+    if hasattr(credential, "api_key"):
+        credential_policy = CognitiveServicesCredentialPolicy(credential)
+    elif credential is not None and not hasattr(credential, "get_token"):
+        raise TypeError("Unsupported credential: {}. Use an instance of TextAnalyticsApiKeyCredential "
+                        "or a token credential from azure.identity".format(type(credential)))
+    return credential_policy
 
-        config = self._create_configuration(**kwargs)
-        config.transport = kwargs.get("transport")  # type: ignore
-        if not config.transport:
-            config.transport = RequestsTransport(**kwargs)
 
-        policies = [
-            config.headers_policy,
-            config.user_agent_policy,
-            RequestIdPolicy(**kwargs),
-            config.proxy_policy,
-            config.redirect_policy,
-            config.retry_policy,
-            credential_policy,
-            config.logging_policy,
-            TextAnalyticsResponseHookPolicy(**kwargs),
-            DistributedTracingPolicy(**kwargs),
-            HttpLoggingPolicy(**kwargs),
-        ]
-        return Pipeline(config.transport, policies=policies)
-
-    def _create_configuration(self, **kwargs):  # pylint: disable=no-self-use
-        config = Configuration(**kwargs)
-        config.user_agent_policy = kwargs.get("user_agent_policy") or \
-            UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
-        config.headers_policy = kwargs.get("headers_policy") or HeadersPolicy(**kwargs)
-        config.proxy_policy = kwargs.get("proxy_policy") or ProxyPolicy(**kwargs)
-        config.logging_policy = kwargs.get("logging_policy") or NetworkTraceLoggingPolicy(**kwargs)
-        config.retry_policy = kwargs.get("retry_policy") or RetryPolicy(**kwargs)
-        config.redirect_policy = kwargs.get("redirect_policy") or RedirectPolicy(**kwargs)
+class TextAnalyticsClientBase(object):
+    def __init__(self, endpoint, credential, aio, **kwargs):
+        if aio:
+            from ._generated.aio import TextAnalyticsClient as Client
+            from .aio._policies_async import AsyncTextAnalyticsResponseHookPolicy as ResponseHookPolicy
+        else:
+            from ._generated import TextAnalyticsClient as Client
+            from ._policies import TextAnalyticsResponseHookPolicy as ResponseHookPolicy
+        self._client = Client(
+            endpoint=endpoint,
+            credential=credential,
+            sdk_moniker=USER_AGENT,
```

Comment:
Shouldn't this produce the right user-agent without passing it USER-AGENT? I think we can delete user_agent.py as well.

Reply:
The reason why I left user agent is because you're including the version of the wrapped client's version. If we want that in the user agent, we still have to pass it in ourself, if not i can get rid of it

Reply:
Adding this telemtry policy link: https://github.com/Azure/azure-sdk/blob/6175e17f2b1077f2a2bcdb73bbc4e93bc5c0d8ad/docs/general/azurecore.md#telemetry-policy. I think we should still override the user agent